### ISSUE TITLE
py2exe does not run from x86 VS prompt on a x64 system

### ISSUE THREAD

**Issue by youca04** [py2exe/py2exe#145]
Python 3.9.10
pip 22.2.2 from C:\Users\xxx\AppData\Local\Programs\Python\Python39\lib\site-packages\pip (python 3.9)
Windows 10 21H2 19044,1889

py2exe was failing for me with even the simplest 'hello world' example, erroring out with file not found for run-py3.9-win32.exe

Similar to this issue in pip - https://github.com/pypa/pip/issues/9962 - I found that runtime.py was using get_platform().

`run_stub = 'run-py%s.%s-%s.exe' % (sys.version_info[0], sys.version_info[1], get_platform())`

After changing that file to use get_host_platform() in all cases, I managed to successfully build a standalone binary for 64-bit.

python -c "import distutils.util; print(distutils.util.get_platform())"
win32

python -c "import distutils.util; print(distutils.util.get_host_platform())"
win-amd64

Regards,

Carl

**Comment by albertosottile** [comment]
Hi, thanks for reporting this. I assume that, like in https://github.com/pypa/pip/issues/9962, you are using a Visual Studio development console. Could you please paste here the output of `set vs`, just to confirm that?

**Comment by youca04** [comment]
Hi Albert, As requested:

```
D:\xxx\>set vs
VS160COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\
VSCMD_ARG_app_plat=Desktop
VSCMD_ARG_HOST_ARCH=x86
VSCMD_ARG_TGT_ARCH=x86
VSCMD_VER=16.11.7
VSINSTALLDIR=C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\
```
And the other output:
```
D:\xxx>python -c "import distutils.util; print(distutils.util.get_platform())"
win32

D:\xxx>python -c "import distutils.util; print(distutils.util.get_host_platform())"
win-amd64

D:\xxx>python -c "import sysconfig; print(sysconfig.get_platform())"
win-amd64
```
Just for completeness - without patching:
```
Building 'dist\er-send-z1.exe'.
error: [Errno 2] No such file or directory: 'C:\\Users\\abc\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\py2exe\\run-py3.9-win32.exe'
```
And with:
```
Building 'dist\er-send-z1.exe'.
Building shared code archive 'dist\library.zip'.
```

**Comment by albertosottile** [comment]
Thanks for the info you provided. The upcoming major of `py2exe` will fix this issue by abandoning `distutils` entirely and using `sysconfig.get_platform`, which we now know returns the correct value. In the meantime, I see you already have a workaround, and so if you do not mind, I would not prepare a temporary patch.

**Comment by youca04** [comment]
Absolutely fine with me. Thanks for a rapid response.

**Comment by albertosottile** [comment]
Fixed in 0.12.0.0
